modules/pom.xml

```diff
@@ -20,10 +20,24 @@
 
     <modules>
         <module>core</module>
-        <module>example</module>
         <module>io</module>
-        <module>tests</module>
-        <module>app</module>
     </modules>
-    
+
+    <profiles>
+        <profile>
+            <id>all</id>
+            <activation>
+                <property>
+                    <name>!release</name>
```

Comment:
Looks good, I usually document profiles and switches like this in comments in the root pom.xml (or at the build instructions)